app/lib/search/backend.dart

```diff
@@ -101,15 +106,28 @@ class SearchBackend {
       '$runtimeVersion/search/update-snapshot',
       expiration: Duration(minutes: 20),
     );
+    final started = clock.now();
     while (true) {
       try {
         await lock.withClaim((claim) async {
-          await doCreateAndUpdateSnapshot(claim);
+          // Force timeout exception if the process does not release the lock.
+          final lockHoldTimeout = _maxLockHoldPeriod + Duration(hours: 1);
+          await doCreateAndUpdateSnapshot(claim).timeout(lockHoldTimeout);
         });
       } catch (e, st) {
-        _logger.warning('Snapshot update failed.', e, st);
+        _logger.pubNoticeShout(
+            'snapshot-building', 'Snapshot update failed.', e, st);
+        // Force waiting at least an hour before we rethrow the exception
```

Comment:
Is this to avoid too quick cycling of restarts? Could you explain that in this comment?